test_db.py

```diff
@@ -0,0 +1,231 @@
+        self.user_id = user.get_user_id(self.cursor, "test@email")
+
+        # prepare the picture_set and picture
+        self.image = Image.new("RGB", (1980, 1080), "blue")
+        self.image_byte_array = io.BytesIO()
+        self.image.save(self.image_byte_array, format="TIFF")
+        self.pic_encoded = base64.b64encode(self.image_byte_array.getvalue()).decode(
+            "utf8"
+        )
+        self.picture_set = picture_set_data.build_picture_set(self.user_id, 1)
+        self.picture = picture_data.build_picture(
+            self.pic_encoded, self.seed_name, 1, 1.0
+        )
+
+    def tearDown(self):
+        db.end_query(self.con, self.cursor)
+
+    def test_new_picture_set(self):
+        """
+        This test checks if the new_picture_set function returns a valid UUID
+        """
+        picture_set_id = picture.new_picture_set(
+            self.cursor, self.picture_set, self.user_id
+        )
+
+        self.assertTrue(
+            is_valid_uuid(picture_set_id), "The picture_set_id is not a valid UUID"
+        )
+
+        self.con.rollback()
+
+    def test_new_picture(self):
+        """
+        This test checks if the new_picture function returns a valid UUID
+        """
+        # prepare the picture_set
+        picture_set_id = picture.new_picture_set(
+            self.cursor, self.picture_set, self.user_id
+        )
+
+        # create the new picture in the db
+        picture_id = picture.new_picture(
+            self.cursor, self.picture, picture_set_id, self.seed_id
+        )
+
+        self.assertTrue(is_valid_uuid(picture_id), "The picture_id is not a valid UUID")
+
+        self.con.rollback()
+
+    def test_get_picture_set(self):
+        """
+        This test checks if the get_picture_set function returns an object
+        """
+        # prepare picture_set
+        picture_set_id = picture.new_picture_set(
+            self.cursor, self.picture_set, self.user_id
+        )
+        # test the function
+        picture_set = picture.get_picture_set(self.cursor, picture_set_id)
+
+        self.assertIsNotNone(picture_set, "The picture_set is None")
+        self.assertNotEqual(len(picture_set), 0, "The picture_set is empty")
+
+        self.con.rollback()
+
+    def test_get_picture(self):
+        """
+        This test checks if the get_picture function returns an object
+        """
+        # prepare picture_set
+        picture_set_id = picture.new_picture_set(
+            self.cursor, self.picture_set, self.user_id
+        )
+
+        # prepare picture
+        picture_id = picture.new_picture(
+            self.cursor, self.picture, picture_set_id, self.seed_id
+        )
+
+        # test the function
+        picture_data = picture.get_picture(self.cursor, picture_id)
+
+        self.assertIsNotNone(picture_data, "The picture_data is None")
+        self.assertNotEqual(len(picture_data), 0, "The picture_data is empty")
+
+        self.con.rollback()
```

Comment:
I see that you use rollback for all your test. Maybe mocking function value would be better for your test, otherwise, you are creating request to the database that might not be necessary.